// shaders/dist/core/Exposure-CFn2J_t3.js
import { vec3, vec4 } from "three/tsl";
const componentDefinition = {
	name: "Exposure",
	category: "Adjustments",
	description: "Multiplicative exposure (gain) on the child. Unlike additive brightness, it scales values around black — blacks stay black while highlights can be pushed past 1.0 to feed HDR bloom and tone-mapping roll-off. Pointwise, so no extra render pass is needed.",
	requiresChild: true,
	props: { exposure: {
		default: 1,
		description: "Brightness multiplier (gain). 1 = unchanged, >1 pushes highlights past white for HDR bloom, <1 darkens.",
		ui: {
			type: ["range", "map"],
			min: 0,
			max: 8,
			step: .1,
			label: "Exposure",
			group: "Appearance"
		}
	} },
	fragmentNode: ({ uniforms, childNode }) => {
		if (!childNode) {
			console.error("You must pass a child component into the Exposure shader.");
			return vec4(0);
		}
		const gain = uniforms.exposure.uniform;
		return vec4(vec3(childNode.r, childNode.g, childNode.b).mul(gain), childNode.a);
	}
};
var Exposure_default = componentDefinition;
export { componentDefinition as n, Exposure_default as t };
