// shaders/dist/core/shaderRegistry-RdJlP7Pq.js
import { n as componentDefinition } from "./AngularBlur-C1oRJ8Y9.js";
import { n as componentDefinition$1 } from "./Ascii-BwSMmD6Z.js";
import { n as componentDefinition$2 } from "./Aurora-D5Pjk4ve.js";
import { n as componentDefinition$3 } from "./BarShift-2XBhva1A.js";
import { n as componentDefinition$4 } from "./Beam-B3BX08lR.js";
import { n as componentDefinition$5 } from "./Blob-BT50tDXE.js";
import { n as componentDefinition$6 } from "./Blur-LkvIPGj0.js";
import { n as componentDefinition$7 } from "./BrickPattern-BfTsjp4I.js";
import { n as componentDefinition$8 } from "./BrightnessContrast-GppIqtkO.js";
import { n as componentDefinition$9 } from "./Bulge-DMOATSEs.js";
import { n as componentDefinition$10 } from "./ChannelBlur-WTXL0B7t.js";
import { n as componentDefinition$11 } from "./Checkerboard-BVBGTvd-.js";
import { n as componentDefinition$12 } from "./Chevron-BKZjAXR_.js";
import { n as componentDefinition$13 } from "./ChromaFlow-C0guw_LP.js";
import { n as componentDefinition$14 } from "./ChromaticAberration-DZOcs-bZ.js";
import { n as componentDefinition$15 } from "./Circle-BycqVCMt.js";
import { n as componentDefinition$16 } from "./ColorWheel-BQUKQPM-.js";
import { n as componentDefinition$17 } from "./ConcentricSpin-CO3Kx1I1.js";
import { n as componentDefinition$18 } from "./ConicGradient-DRvVwiSj.js";
import { n as componentDefinition$19 } from "./ContourLines-s4waN3uU.js";
import { n as componentDefinition$20 } from "./Crescent-ppz7W7eM.js";
import { n as componentDefinition$21 } from "./Cross-BfNOtQ6s.js";
import { n as componentDefinition$22 } from "./CRTScreen-corekL_R.js";
import { n as componentDefinition$23 } from "./Crystal-QsQoy8fZ.js";
import { n as componentDefinition$24 } from "./CursorRipples-CLP7Sinv.js";
import { n as componentDefinition$25 } from "./CursorTrail-CshoSM5Y.js";
import { n as componentDefinition$26 } from "./DiamondGradient-D6eyjZOn.js";
import { n as componentDefinition$27 } from "./DiffuseBlur-BjvXHmJx.js";
import { n as componentDefinition$28 } from "./Dither-z6s-TFg7.js";
import { n as componentDefinition$29 } from "./DOMTexture-Ec0k6AnK.js";
import { n as componentDefinition$30 } from "./DotGrid-BxAAGO4F.js";
import { n as componentDefinition$31 } from "./DropShadow-CR3cecIQ.js";
import { n as componentDefinition$32 } from "./Duotone-PwTfBBtM.js";
import { n as componentDefinition$33 } from "./Ellipse-sWhNvW9-.js";
import { n as componentDefinition$34 } from "./Emboss-DGaubb9x.js";
import { n as componentDefinition$35 } from "./Exposure-CFn2J_t3.js";
import { n as componentDefinition$36 } from "./FallingLines-9elMmSlj.js";
import { n as componentDefinition$37 } from "./FilmGrain-DIydVgOW.js";
import { n as componentDefinition$38 } from "./FloatingParticles-B068TLIi.js";
import { n as componentDefinition$39 } from "./Flower-DSr0JWa2.js";
import { n as componentDefinition$40 } from "./FlowField-CeZjhyEo.js";
import { n as componentDefinition$41 } from "./FlowingGradient-Bd8KBK5P.js";
import { n as componentDefinition$42 } from "./FlutedGlass-BJtMfV2c.js";
import { n as componentDefinition$43 } from "./Fog-BQuVDReH.js";
import { n as componentDefinition$44 } from "./Form3D-DOek31CJ.js";
import { n as componentDefinition$45 } from "./FractalNoise-Dyhbvx9a.js";
import { n as componentDefinition$46 } from "./Glass-Cy1wvfuu.js";
import { n as componentDefinition$47 } from "./GlassTiles-Bb3cuOkr.js";
import { n as componentDefinition$48 } from "./Glitch-zqdbCc2E.js";
import { n as componentDefinition$49 } from "./Glow-CJ7MTzc4.js";
import { n as componentDefinition$50 } from "./Godrays-DdSLZtBs.js";
import { n as componentDefinition$51 } from "./Grayscale-BC40hfXH.js";
import { n as componentDefinition$52 } from "./Grid-Lc4S0Hmp.js";
import { n as componentDefinition$53 } from "./GridDistortion-DQcPYbXS.js";
import { n as componentDefinition$54 } from "./Group-BeZhHbOT.js";
import { n as componentDefinition$55 } from "./Halftone-CsblgxnY.js";
import { n as componentDefinition$56 } from "./HexGrid-DWRNPMET.js";
import { n as componentDefinition$57 } from "./HueShift-C-p9W5sr.js";
import { n as componentDefinition$58 } from "./ImageTexture-bqgQxAt9.js";
import { n as componentDefinition$59 } from "./Invert-D-RVP34v.js";
import { n as componentDefinition$60 } from "./Kaleidoscope-HQNQrRP_.js";
import { n as componentDefinition$61 } from "./LensFlare-DtES749p.js";
import { n as componentDefinition$62 } from "./LinearBlur-6mCJ9fY5.js";
import { n as componentDefinition$63 } from "./LinearGradient-DIbCAHc7.js";
import { n as componentDefinition$64 } from "./Liquify-8dcdv2Sl.js";
import { n as componentDefinition$65 } from "./Marble-DX29RG-p.js";
import { n as componentDefinition$66 } from "./Mirror-BM2xf3Nr.js";
import { n as componentDefinition$67 } from "./MultiPointGradient-BzM24vnR.js";
import { n as componentDefinition$68 } from "./Neon-nPiMi3wt.js";
import { n as componentDefinition$69 } from "./Paper-Ddpe8-EI.js";
import { n as componentDefinition$70 } from "./Perspective-BIKpqd4H.js";
import { n as componentDefinition$71 } from "./Pixelate-Cp0-lwkG.js";
import { n as componentDefinition$72 } from "./Plasma-M--gLd1Q.js";
import { n as componentDefinition$73 } from "./PolarCoordinates-C42rlD2q.js";
import { n as componentDefinition$74 } from "./Polygon-CKqH0ApY.js";
import { n as componentDefinition$75 } from "./Posterize-B6ZhRkH2.js";
import { n as componentDefinition$76 } from "./ProgressiveBlur-jvN5Adf4.js";
import { n as componentDefinition$77 } from "./RadialGradient-SHxJ9ZNF.js";
import { n as componentDefinition$78 } from "./RectangularCoordinates-Dw6Xr9uV.js";
import { n as componentDefinition$79 } from "./ReflectivePlane-CGXUsPVJ.js";
import { n as componentDefinition$80 } from "./Ring-V4sEHwwm.js";
import { n as componentDefinition$81 } from "./Ripples-DC1y3YuP.js";
import { n as componentDefinition$82 } from "./RoundedRect-JY2z6YEH.js";
import { n as componentDefinition$83 } from "./Saturation-ChJ9IWnl.js";
import { n as componentDefinition$84 } from "./Sharpness-C1ufX8S-.js";
import { n as componentDefinition$85 } from "./Shatter-Du5THcwZ.js";
import { n as componentDefinition$86 } from "./SimplexNoise--vwIZVu0.js";
import { n as componentDefinition$87 } from "./SineWave-BIVPG3oc.js";
import { n as componentDefinition$88 } from "./Smoke-DSBQreo0.js";
import { n as componentDefinition$89 } from "./SmokeFill-Dy-0kjsX.js";
import { n as componentDefinition$90 } from "./Solarize-BymZlSu0.js";
import { n as componentDefinition$91 } from "./SolidColor-D8XC4aoJ.js";
import { n as componentDefinition$92 } from "./Spherize-tRTB4WdO.js";
import { n as componentDefinition$93 } from "./Spiral-CiDp2Tg2.js";
import { n as componentDefinition$94 } from "./Star-Cp4dt6Fl.js";
import { n as componentDefinition$95 } from "./Strands-V3q8N-y1.js";
import { n as componentDefinition$96 } from "./Stretch-CTgtc3nG.js";
import { n as componentDefinition$97 } from "./Stripes-CZV6oGMy.js";
import { n as componentDefinition$98 } from "./StudioBackground-CWSHq6vE.js";
import { n as componentDefinition$99 } from "./SunBurst-CskZwXc3.js";
import { n as componentDefinition$100 } from "./Swirl-CyGfVZ8n.js";
import { n as componentDefinition$101 } from "./ThinFilm-CUsi8eSg.js";
import { n as componentDefinition$102 } from "./TiltShift-e-1ia_HN.js";
import { n as componentDefinition$103 } from "./Tint-R7y5CJ4U.js";
import { n as componentDefinition$104 } from "./Trapezoid-BXzH--z9.js";
import { n as componentDefinition$105 } from "./Tritone-CF4Oj1sl.js";
import { n as componentDefinition$106 } from "./Truchet-BL4jzZDX.js";
import { n as componentDefinition$107 } from "./Twirl-BVLPHKJD.js";
import { n as componentDefinition$108 } from "./Vesica-DzIEa3aJ.js";
import { n as componentDefinition$109 } from "./VHS-B2IBdQgX.js";
import { n as componentDefinition$110 } from "./Vibrance-jrvT49Du.js";
import { n as componentDefinition$111 } from "./VideoTexture-FmV1l91c.js";
import { n as componentDefinition$112 } from "./Vignette-ng_5gjXc.js";
import { n as componentDefinition$113 } from "./Voronoi-C8iIU7AT.js";
import { n as componentDefinition$114 } from "./WaveDistortion-Crt9JSXN.js";
import { n as componentDefinition$115 } from "./Weave-CFmLkEaq.js";
import { n as componentDefinition$116 } from "./WebcamTexture-DlS4MCDm.js";
import { n as componentDefinition$117 } from "./WorleyNoise-CEZiaVAN.js";
import { n as componentDefinition$118 } from "./ZoomBlur-DTVSwIvc.js";
var createShaderRegistry = () => {
	const registry = {};
	const shaderDefinitions = {
		AngularBlur: componentDefinition,
		Ascii: componentDefinition$1,
		Aurora: componentDefinition$2,
		BarShift: componentDefinition$3,
		Beam: componentDefinition$4,
		Blob: componentDefinition$5,
		Blur: componentDefinition$6,
		BrickPattern: componentDefinition$7,
		BrightnessContrast: componentDefinition$8,
		Bulge: componentDefinition$9,
		ChannelBlur: componentDefinition$10,
		Checkerboard: componentDefinition$11,
		Chevron: componentDefinition$12,
		ChromaFlow: componentDefinition$13,
		ChromaticAberration: componentDefinition$14,
		Circle: componentDefinition$15,
		ColorWheel: componentDefinition$16,
		ConcentricSpin: componentDefinition$17,
		ConicGradient: componentDefinition$18,
		ContourLines: componentDefinition$19,
		Crescent: componentDefinition$20,
		Cross: componentDefinition$21,
		CRTScreen: componentDefinition$22,
		Crystal: componentDefinition$23,
		CursorRipples: componentDefinition$24,
		CursorTrail: componentDefinition$25,
		DiamondGradient: componentDefinition$26,
		DiffuseBlur: componentDefinition$27,
		Dither: componentDefinition$28,
		DOMTexture: componentDefinition$29,
		DotGrid: componentDefinition$30,
		DropShadow: componentDefinition$31,
		Duotone: componentDefinition$32,
		Ellipse: componentDefinition$33,
		Emboss: componentDefinition$34,
		Exposure: componentDefinition$35,
		FallingLines: componentDefinition$36,
		FilmGrain: componentDefinition$37,
		FloatingParticles: componentDefinition$38,
		Flower: componentDefinition$39,
		FlowField: componentDefinition$40,
		FlowingGradient: componentDefinition$41,
		FlutedGlass: componentDefinition$42,
		Fog: componentDefinition$43,
		Form3D: componentDefinition$44,
		FractalNoise: componentDefinition$45,
		Glass: componentDefinition$46,
		GlassTiles: componentDefinition$47,
		Glitch: componentDefinition$48,
		Glow: componentDefinition$49,
		Godrays: componentDefinition$50,
		Grayscale: componentDefinition$51,
		Grid: componentDefinition$52,
		GridDistortion: componentDefinition$53,
		Group: componentDefinition$54,
		Halftone: componentDefinition$55,
		HexGrid: componentDefinition$56,
		HueShift: componentDefinition$57,
		ImageTexture: componentDefinition$58,
		Invert: componentDefinition$59,
		Kaleidoscope: componentDefinition$60,
		LensFlare: componentDefinition$61,
		LinearBlur: componentDefinition$62,
		LinearGradient: componentDefinition$63,
		Liquify: componentDefinition$64,
		Marble: componentDefinition$65,
		Mirror: componentDefinition$66,
		MultiPointGradient: componentDefinition$67,
		Neon: componentDefinition$68,
		Paper: componentDefinition$69,
		Perspective: componentDefinition$70,
		Pixelate: componentDefinition$71,
		Plasma: componentDefinition$72,
		PolarCoordinates: componentDefinition$73,
		Polygon: componentDefinition$74,
		Posterize: componentDefinition$75,
		ProgressiveBlur: componentDefinition$76,
		RadialGradient: componentDefinition$77,
		RectangularCoordinates: componentDefinition$78,
		ReflectivePlane: componentDefinition$79,
		Ring: componentDefinition$80,
		Ripples: componentDefinition$81,
		RoundedRect: componentDefinition$82,
		Saturation: componentDefinition$83,
		Sharpness: componentDefinition$84,
		Shatter: componentDefinition$85,
		SimplexNoise: componentDefinition$86,
		SineWave: componentDefinition$87,
		Smoke: componentDefinition$88,
		SmokeFill: componentDefinition$89,
		Solarize: componentDefinition$90,
		SolidColor: componentDefinition$91,
		Spherize: componentDefinition$92,
		Spiral: componentDefinition$93,
		Star: componentDefinition$94,
		Strands: componentDefinition$95,
		Stretch: componentDefinition$96,
		Stripes: componentDefinition$97,
		StudioBackground: componentDefinition$98,
		SunBurst: componentDefinition$99,
		Swirl: componentDefinition$100,
		ThinFilm: componentDefinition$101,
		TiltShift: componentDefinition$102,
		Tint: componentDefinition$103,
		Trapezoid: componentDefinition$104,
		Tritone: componentDefinition$105,
		Truchet: componentDefinition$106,
		Twirl: componentDefinition$107,
		Vesica: componentDefinition$108,
		VHS: componentDefinition$109,
		Vibrance: componentDefinition$110,
		VideoTexture: componentDefinition$111,
		Vignette: componentDefinition$112,
		Voronoi: componentDefinition$113,
		WaveDistortion: componentDefinition$114,
		Weave: componentDefinition$115,
		WebcamTexture: componentDefinition$116,
		WorleyNoise: componentDefinition$117,
		ZoomBlur: componentDefinition$118
	};
	Object.entries(shaderDefinitions).forEach(([fileName, def]) => {
		const propsMetadata = {};
		if (def.props) Object.entries(def.props).forEach(([propName, propConfig]) => {
			if (propConfig.ui) propsMetadata[propName] = {
				ui: propConfig.ui,
				default: propConfig.default,
				description: propConfig.description
			};
		});
		registry[def.name] = {
			name: def.name,
			fileName,
			category: def.category || "Uncategorized",
			definition: def,
			propsMetadata
		};
	});
	return registry;
};
const shaderRegistry = createShaderRegistry();
function getAllShaders() {
	return Object.values(shaderRegistry);
}
function getShaderByName(name) {
	return shaderRegistry[name];
}
function getShadersByCategory(category) {
	return Object.values(shaderRegistry).filter((shader) => shader.category === category);
}
function getShaderCategories() {
	const categories = new Set(Object.values(shaderRegistry).map((shader) => shader.category));
	return Array.from(categories).sort();
}
export { shaderRegistry as a, getShadersByCategory as i, getShaderByName as n, getShaderCategories as r, getAllShaders as t };
