// shaders/dist/core/ThinFilm-CUsi8eSg.js
import { a as transformColor, l as transformPosition, o as transformColorSpace, t as colorSpaceOptions } from "./transformations-CJcUeZIC.js";
import { t as mixColors } from "./colorMixing-BPpDnR5I.js";
import { t as createAnimatedTime } from "./time-DUqSFWvT.js";
import { n as createAnalyticSdfSampler, r as createSvgSdfSampler } from "./sdf-D_EAGXhN.js";
import { Fn, clamp, cos, float, floor, fract, mix, screenUV, sin, sqrt, step, uniform, vec2, vec3, vec4, viewportSize } from "three/tsl";
var TWO_PI = Math.PI * 2;
function applyThinFilmEffect(sdfSampler, params) {
	const uniforms = params.uniforms;
	const lx = uniform(0);
	const ly = uniform(0);
	const updateLightAngle = () => {
		const rad = uniforms.lightAngle.uniform.value * Math.PI / 180;
		lx.value = Math.cos(rad);
		ly.value = Math.sin(rad);
	};
	updateLightAngle();
	params.onBeforeRender(updateLightAngle);
	const animTime = createAnimatedTime(params, uniforms.speed);
	const mode = uniforms.mode.uniform.value;
	return Fn(() => {
		const uv$1 = screenUV;
		const aspect = viewportSize.x.div(viewportSize.y);
		const sharp = uniforms.edgeSoftness.uniform.mul(float(.5)).max(float(.001));
		const centerPos = vec2(uniforms.center.uniform.x, uniforms.center.uniform.y.oneMinus());
		const scale = uniforms.scale.uniform;
		const sdfUV = vec2(uv$1.x.sub(centerPos.x).mul(aspect).div(scale).add(float(.5)), uv$1.y.sub(centerPos.y).div(scale).add(float(.5)));
		const sdfRaw = sdfSampler(sdfUV).r;
		const sdf = sdfRaw.div(scale);
		const EPS = float(.01);
		const gradX = sdfSampler(sdfUV.add(vec2(EPS, float(0)))).r.sub(sdfRaw).div(EPS);
		const gradY = sdfSampler(sdfUV.add(vec2(float(0), EPS))).r.sub(sdfRaw).div(EPS);
		const gLen = sqrt(gradX.mul(gradX).add(gradY.mul(gradY))).max(float(1e-4));
		const nx = gradX.div(gLen);
		const ny = gradY.div(gLen);
		const rb1 = clamp(sdf.negate().div(sharp).mul(float(32)), float(0), float(1));
		const rimWidth = uniforms.rimWidth.uniform.mul(float(.1)).max(float(.001));
		const depth = clamp(sdf.negate().div(rimWidth), float(0), float(1));
		const rim = float(1).sub(depth).mul(float(1).sub(depth)).mul(rb1);
		const baseCos = nx.mul(lx).add(ny.mul(ly));
		const baseSin = ny.mul(lx).sub(nx.mul(ly));
		const theta = animTime.mul(float(TWO_PI));
		const rotatedDot = baseCos.mul(cos(theta)).add(baseSin.mul(sin(theta)));
		const phase = uniforms.hueShift.uniform.add(uniforms.dispersion.uniform.mul(rotatedDot)).add(uniforms.thickness.uniform.mul(depth).mul(float(2)));
		let baseColor;
		if (mode === "custom") {
			const t3 = fract(phase).mul(float(3));
			const si = fract(t3);
			const floorT3 = floor(t3);
			const cs = uniforms.colorSpace.uniform.value;
			const c01 = mixColors(uniforms.colorA.uniform, uniforms.colorB.uniform, si, cs);
			const c12 = mixColors(uniforms.colorB.uniform, uniforms.colorC.uniform, si, cs);
			let res = mixColors(uniforms.colorC.uniform, uniforms.colorA.uniform, si, cs);
			res = mix(res, c12, step(floorT3, float(1.5)));
			res = mix(res, c01, step(floorT3, float(.5)));
			baseColor = res.rgb;
		} else baseColor = vec3(cos(phase.add(float(0)).mul(float(TWO_PI))), cos(phase.add(float(.3333)).mul(float(TWO_PI))), cos(phase.add(float(.6667)).mul(float(TWO_PI)))).mul(float(.5)).add(float(.5));
		return vec4(mix(vec3(1), baseColor, uniforms.saturation.uniform).mul(rim).mul(uniforms.intensity.uniform), clamp(rim, float(0), float(1)));
	})();
}
var DEFAULT_SHAPE_CONFIG = JSON.stringify({
	type: "circleSDF",
	radius: .35
});
const componentDefinition = {
	name: "ThinFilm",
	category: "Shape Effects",
	description: "Iridescent thin-film edge",
	props: {
		center: {
			default: {
				x: .5,
				y: .5
			},
			transform: transformPosition,
			description: "Center position of the shape",
			ui: {
				type: "position",
				label: "Center",
				group: "Position"
			}
		},
		scale: {
			default: 1,
			description: "Scale of the shape (1 = default size)",
			ui: {
				type: ["range", "map"],
				min: .1,
				max: 3,
				step: .05,
				label: "Scale",
				group: "Position"
			}
		},
		intensity: {
			default: 1,
			description: "Rim brightness — values above 1 push the edge past white for HDR bloom",
			ui: {
				type: ["range", "map"],
				min: 0,
				max: 4,
				step: .05,
				label: "Intensity",
				group: "Film"
			}
		},
		rimWidth: {
			default: 1,
			description: "How far inward from the edge the iridescence reaches",
			ui: {
				type: ["range", "map"],
				min: 0,
				max: 3,
				step: .01,
				label: "Rim Width",
				group: "Film"
			}
		},
		edgeSoftness: {
			default: .3,
			description: "Softness of the shape boundary",
			ui: {
				type: ["range", "map"],
				min: 0,
				max: 1,
				step: .05,
				label: "Edge Softness",
				group: "Film"
			}
		},
		thickness: {
			default: .5,
			description: "Film thickness — higher values pack more spectral bands across the rim",
			ui: {
				type: ["range", "map"],
				min: 0,
				max: 1,
				step: .01,
				label: "Thickness",
				group: "Film"
			}
		},
		dispersion: {
			default: .5,
			description: "How strongly colour separates around the perimeter (warm vs cool sides)",
			ui: {
				type: ["range", "map"],
				min: 0,
				max: 1,
				step: .01,
				label: "Dispersion",
				group: "Film"
			}
		},
		saturation: {
			default: 1,
			description: "Spectral saturation — 0 = white rim, 1 = full iridescence",
			ui: {
				type: ["range", "map"],
				min: 0,
				max: 1,
				step: .01,
				label: "Saturation",
				group: "Film"
			}
		},
		hueShift: {
			default: 0,
			description: "Rotate the spectrum to reposition the colours",
			ui: {
				type: ["range", "map"],
				min: 0,
				max: 1,
				step: .01,
				label: "Hue Shift",
				group: "Film"
			}
		},
		lightAngle: {
			default: 300,
			description: "Light angle in degrees — sets which side of the rim runs warm vs cool",
			ui: {
				type: "range",
				min: 0,
				max: 360,
				step: 1,
				label: "Light Angle",
				group: "Film"
			}
		},
		mode: {
			default: "rainbow",
			compileTime: true,
			description: "Rainbow uses the full iridescent spectrum; Custom cycles through your three chosen colours",
			ui: {
				type: "select",
				options: [{
					label: "Rainbow",
					value: "rainbow"
				}, {
					label: "Custom",
					value: "custom"
				}],
				label: "Mode",
				group: "Colors"
			}
		},
		colorA: {
			default: "#2b6fff",
			transform: transformColor,
			description: "First colour in the rim cycle",
			ui: {
				type: "color",
				label: "Color 1",
				group: "Colors",
				condition: { mode: "custom" }
			}
		},
		colorB: {
			default: "#ffffff",
			transform: transformColor,
			description: "Second colour in the rim cycle",
			ui: {
				type: "color",
				label: "Color 2",
				group: "Colors",
				condition: { mode: "custom" }
			}
		},
		colorC: {
			default: "#ff7a21",
			transform: transformColor,
			description: "Third colour in the rim cycle",
			ui: {
				type: "color",
				label: "Color 3",
				group: "Colors",
				condition: { mode: "custom" }
			}
		},
		colorSpace: {
			default: "oklch",
			transform: transformColorSpace,
			compileTime: true,
			description: "Colour space used to blend between custom colours",
			ui: {
				type: "select",
				options: colorSpaceOptions,
				label: "Color Space",
				group: "Colors",
				condition: { mode: "custom" }
			}
		},
		speed: {
			default: .1,
			description: "Speed at which the colours rotate around the rim (0 = static)",
			ui: {
				type: "range",
				min: -1,
				max: 1,
				step: .01,
				label: "Speed",
				group: "Animation"
			}
		},
		shape: {
			default: DEFAULT_SHAPE_CONFIG,
			description: "Serialized shape configuration (JSON)",
			ui: {
				type: "shape",
				label: "Shape",
				group: "Shape"
			}
		},
		shapeSdfUrl: {
			default: "",
			compileTime: true,
			description: "URL to a pre-generated SDF .bin file — when non-empty, activates SVG mode and triggers a shader recompile"
		},
		shapeType: {
			default: "",
			compileTime: true,
			description: "Active SDF shape type — triggers recompile when shape is switched. When empty, derived from shape JSON at mount time."
		}
	},
	fragmentNode: (params) => {
		const { uniforms, onCleanup, onBeforeRender } = params;
		const shapeSdfUrl = uniforms.shapeSdfUrl.uniform.value;
		const initialConfig = (() => {
			const val = uniforms.shape.uniform.value;
			if (typeof val === "object" && val !== null) return val;
			try {
				return JSON.parse(val);
			} catch {
				return {
					type: "circleSDF",
					radius: .35
				};
			}
		})();
		if (shapeSdfUrl) return applyThinFilmEffect(createSvgSdfSampler(shapeSdfUrl, onBeforeRender, onCleanup), params);
		else return applyThinFilmEffect(createAnalyticSdfSampler(uniforms.shapeType.uniform.value || initialConfig.type || "circleSDF", initialConfig, uniforms.shape, onBeforeRender), params);
	}
};
var ThinFilm_default = componentDefinition;
export { componentDefinition as n, ThinFilm_default as t };
